### PR TITLE
Debug Query as SQL string. 

### LINKED ISSUE #146: Implement `std::fmt::Display` for `Query`
It might be helpful to log the queries. 
Currently, it is not possible:
```
error[E0277]: `Query` doesn't implement `std::fmt::Display`
  |
8 |     println!("{query}");
  |               ^^^^^^^ `Query` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `Query`
```

The current implementation of SQLBuilder moves `mut self,` so it looks like some refactoring will be required there.

### PR DESCRIPTION
**Pull request by pravic**
## Summary

Allow to print the current SQL query, whether it's bound or not. Useful for logging and diagnostics.

Note: I chose `Debug` over `Display` because the latter is too implicit and somebody might accidentally pass a query into a string.

Albeit, something like `fn Query::into_string(self) -> String` can be useful on its own - as a poor man's query builder:

```rust
let q = client.query("SELECT id from system.users WHERE name = ?");
let q = q.bind("default");

// store the query string for later 
let query_string = q.into_string(); // returns `q.sql.finish()` inernally

// then execute it:
client.query(&query_string).fetch();
```

closes #146 

## Checklist
Delete items not relevant to your PR:
- [x] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided so that we can include it in CHANGELOG later
- [ ] For significant changes, documentation in README and https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials
- [ ] **conflicts** with #154, this PR needs to be rebased on top of that

### CODE REVIEW

**Review by serprex — APPROVED** [review]

**Review by loyd — CHANGES_REQUESTED** [review]
Only misleading `Debug` impl, all other changes are fine

File: src/query.rs [hunk 1]
```diff
@@ -22,6 +23,12 @@ pub struct Query {
     sql: SqlBuilder,
 }
 
+impl fmt::Debug for Query {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{sql}", sql = self.sql)
```
[the remark this answers is not in the document]
Comment (pravic) [reply src/query.rs:28]: Also, `sql_display` or `as_sql`?

[the remark this answers is not in the document]
Comment (pravic) [reply src/query.rs]: @loyd Removed `Debug` in f4da92b.

**Review by loyd — APPROVED** [review]

**Comment by loyd** [comment]
@pravic, thanks for your contribution!

**Pull request merged**